### PR TITLE
Wallet: Refactor wallet name logic

### PR DESCRIPTION
**Pull request by figitaki**
I moved the logic for displaying a wallet name into a component called `WalletName` that renders the wallet name correctly and applies the valid text styles.

### CODE REVIEW

File: frontend/wallet/src/render/views/footer/SendButton.re [hunk 1]
```diff
@@ -142,16 +140,16 @@ let make = (~wallets, ~onSubmit) => {
...
+                        <WalletName pubkey={wallet.key} />
                       )
                     )
                  |> Array.toList
                }
              />
              spacer
              <TextField
-               label="To"
+             label="To"
```
Comment (o1pranay) [inline frontend/wallet/src/render/views/footer/SendButton.re]: nit fix spacing

File: frontend/wallet/src/render/views/settings/SettingsPage.re [hunk 2]
```diff
@@ -81,14 +81,13 @@ module SettingsQuery = ReasonApollo.CreateQuery(SettingsQueryString);
...
   let make = (~publicKey) => {
-    let (addressBook, _) = React.useContext(AddressBookProvider.context);
     let keyStr = PublicKey.toString(publicKey);
     let route = "/settings/" ++ Js.Global.encodeURIComponent(keyStr);
     <div
       className=Styles.walletItem
       onClick={_ => ReasonReact.Router.push(route)}>
       <div className=Styles.walletName>
-        {React.string(AddressBook.getWalletName(addressBook, publicKey))}
+        <WalletName pubkey={publicKey} />
```
Comment (o1pranay) [inline frontend/wallet/src/render/views/settings/SettingsPage.re]: maybe we can rename the prop in <WalletName> publicKey so we can use shorthand notation?

Comment (figitaki) [reply frontend/wallet/src/render/views/settings/SettingsPage.re]: We've decided to rename all publicKey to pubkey

Comment (o1pranay) [reply frontend/wallet/src/render/views/settings/SettingsPage.re]: ok whatever name we want, we can be consistent 
`let make = (~pubkey) => {`

Comment (Schmavery) [reply frontend/wallet/src/render/views/settings/SettingsPage.re]: :'(

**Pull request merged**